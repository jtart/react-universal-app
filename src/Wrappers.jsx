import React from 'react';
import { StaticRouter, BrowserRouter } from 'react-router-dom';
import App from './App.jsx';

export const ClientApp = props => (
  <BrowserRouter {...props}>
    <App initialData={props.data} routes={props.routes} />
  </BrowserRouter>
);

export const ServerApp = props => (
  <StaticRouter {...props}>
    <App initialData={props.data} routes={props.routes} />
  </StaticRouter>
);
