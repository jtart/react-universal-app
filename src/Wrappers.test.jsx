import React from 'react';
import { ClientApp, ServerApp } from './Wrappers';

describe('ClientApp', () => {
  it('should render as exepcted', () => {
    const wrapper = shallow(
      <ClientApp data="someData!" routes={['someRoute']} />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});

describe('ServerApp', () => {
  describe('no passed routerContext', () => {
    it('should render as exepcted', () => {
      const wrapper = shallow(
        <ServerApp
          location="someUrl"
          routes={['someRoute']}
          data="somePassedData"
          context={{}}
        />,
      );

      expect(wrapper).toMatchSnapshot();
    });
  });

  it('should render as exepcted', () => {
    const wrapper = shallow(
      <ServerApp
        location="someUrl"
        routes={['someRoute']}
        data="somePassedData"
        context={{ context: 'someRouterContext' }}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
