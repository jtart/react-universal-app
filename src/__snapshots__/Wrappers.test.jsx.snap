// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientApp should render as exepcted 1`] = `
<BrowserRouter
  data="someData!"
  routes={
    Array [
      "someRoute",
    ]
  }
>
  <withRouter(App)
    initialData="someData!"
    routes={
      Array [
        "someRoute",
      ]
    }
  />
</BrowserRouter>
`;

exports[`ServerApp no passed routerContext should render as exepcted 1`] = `
<StaticRouter
  basename=""
  context={Object {}}
  data="somePassedData"
  location="someUrl"
  routes={
    Array [
      "someRoute",
    ]
  }
>
  <withRouter(App)
    initialData="somePassedData"
    routes={
      Array [
        "someRoute",
      ]
    }
  />
</StaticRouter>
`;

exports[`ServerApp should render as exepcted 1`] = `
<StaticRouter
  basename=""
  context={
    Object {
      "context": "someRouterContext",
    }
  }
  data="somePassedData"
  location="someUrl"
  routes={
    Array [
      "someRoute",
    ]
  }
>
  <withRouter(App)
    initialData="somePassedData"
    routes={
      Array [
        "someRoute",
      ]
    }
  />
</StaticRouter>
`;
